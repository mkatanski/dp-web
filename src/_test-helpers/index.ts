export * from "./withResponsiveContext";
export * from "./withThemeProvider";
