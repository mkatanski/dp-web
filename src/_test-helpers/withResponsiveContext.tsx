import React from "react";
import { Context as ResponsiveContext } from "react-responsive";
import { Breakpoints } from "config/breakpoints";

export const withMobileSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.mobile }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withTabletSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.tablet }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withDesktopSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.desktop }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withLargeDesktopSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.largeDesktop }}>
    {Component}
  </ResponsiveContext.Provider>
);
