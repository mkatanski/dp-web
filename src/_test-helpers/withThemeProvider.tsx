import React from "react";
import { ThemeProvider as DefaultThemeProvider } from "lib/theme/ThemeProvider";
import { StyledTheme, Theme } from "styles/theme/_types";
import { createMuiTheme } from "@material-ui/core";

const testTheme: StyledTheme = {
  primary: "primary",
  secondary: "secondary",
  textPrimary: "textPrimary",
  textSecondary: "textSecondary",
  textInverse: "textInverse"
};

const muiTheme = createMuiTheme({
  direction: "ltr"
});

const theme: Theme = Object.assign(testTheme, muiTheme);

export const withThemeProvider = (Component: React.ReactNode) => (
  <DefaultThemeProvider theme={theme}>{Component}</DefaultThemeProvider>
);
