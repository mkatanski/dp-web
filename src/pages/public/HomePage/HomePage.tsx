import React from "react";
import { PageTitle } from "components/PageTitle";
import { useDeploymentsResource } from "hooks/useDeploymentsResource";

const HomeView: React.FC = () => {
  useDeploymentsResource();
  return <PageTitle category="general" title="Deployments" />;
};

export default HomeView;
