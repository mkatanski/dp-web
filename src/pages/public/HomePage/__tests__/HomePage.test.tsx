/* eslint-disable react/display-name */
import React from "react";
import TestRenderer from "react-test-renderer";
import "jest-styled-components";

import HomePage from "../HomePage";
import { withThemeProvider } from "_test-helpers";

jest.mock("components/PageTitle", () => ({
  PageTitle: ({ ...rest }: React.PropsWithChildren<{}>) => (
    <div id="PageTitle" data-props={rest} />
  )
}));

jest.mock("views/DeploymentsTableView", () => ({
  DeploymentsTableView: ({ ...rest }: React.PropsWithChildren<{}>) => (
    <div id="DeploymentsTableView" data-props={rest} />
  )
}));

jest.mock("components/ToggleableDrawer", () => ({
  ToggleableDrawer: ({ ...rest }: React.PropsWithChildren<{}>) => (
    <div id="ToggleableDrawer" data-props={rest} />
  )
}));

describe("pages", () => {
  describe("protected", () => {
    describe("HomePage", () => {
      it("should render HomePage component", () => {
        const tree = TestRenderer.create(withThemeProvider(<HomePage />));
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
