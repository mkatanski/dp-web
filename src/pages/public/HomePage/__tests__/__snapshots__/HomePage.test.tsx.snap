// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages protected HomePage should render HomePage component 1`] = `
Array [
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

@media (min-width:576px) {
  .c0 {
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }
}

@media (min-width:768px) {
  .c0 {
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }
}

<div
    data-props={
      Object {
        "alignContent": "flex-end",
        "alignItems": "flex-end",
        "container": true,
      }
    }
    id="MUI.Grid"
  >
    <div
      data-props={
        Object {
          "item": true,
          "sm": 6,
          "xs": 12,
        }
      }
      id="MUI.Grid"
    >
      <div
        data-props={
          Object {
            "category": "general",
            "title": "Deployments",
          }
        }
        id="PageTitle"
      />
    </div>
    <div
      data-props={
        Object {
          "alignContent": "flex-end",
          "alignItems": "flex-end",
          "item": true,
          "sm": 6,
          "xs": 12,
        }
      }
      id="MUI.Grid"
    >
      <div
        className="c0"
      >
        <div
          data-props={
            Object {
              "children": <DeploymentForm />,
              "drawerAnchor": "right",
              "renderButton": [Function],
            }
          }
          id="ToggleableDrawer"
        />
      </div>
    </div>
  </div>,
  @media (min-width:576px) {

}

@media (min-width:768px) {

}

<div
    data-props={Object {}}
    id="DeploymentsTableView"
  />,
]
`;
